scripts/lib/CIME/BuildTools/configure.py

```diff
@@ -79,6 +79,31 @@ def configure(
                     new_cmake_macros_dir, os.path.join(output_dir, "cmake_macros")
                 )
 
+            if form == "Makefile":
+                # Use the cmake macros to generate the make macros
```

Comment:
This feel tacky. Fortunately, it appears as though using `configure` to create Makefile macros is very uncommon. With this PR, the only remaining use is in `tools/mapping/gen_domain_files/test_gen_domain.sh`

Reply:
I don't see any reason that we cannot replace this (in  test_gen_domain.sh) with a cmake build - do you?  That would allow us to remove this ```if form```?

Reply:
@jedwards4b , it looks like the source code in `tools/mapping/gen_domain_files/src/*` requires a Makefile build system, so we'd need to change that in order to be able to remove Makefile mode from `configure`.